addons/sale_stock/models/account_move.py

```diff
@@ -25,86 +25,61 @@ def _stock_account_get_last_step_stock_moves(self):
     def _get_invoiced_lot_values(self):
         """ Get and prepare data to show a table of invoiced lot on the invoice's report. """
         self.ensure_one()
-
-        if self.state == 'draft':
+        if self.state == 'draft' or not self.invoice_date or self.type not in ('out_invoice', 'out_refund'):
             return []
 
-        sale_lines = self.invoice_line_ids.sale_line_ids
-        sale_orders = sale_lines.order_id
-        stock_move_lines = sale_lines.move_ids.filtered(lambda r: r.state == 'done').move_line_ids
-
-        # Get the other customer invoices and refunds.
-        ordered_invoice_ids = sale_orders.mapped('invoice_ids')\
-            .filtered(lambda i: i.state not in ['draft', 'cancel'])\
-            .sorted(lambda i: (i.invoice_date, i.id))
-
-        # Get the position of self in other customer invoices and refunds.
-        self_index = None
-        i = 0
-        for invoice in ordered_invoice_ids:
-            if invoice.id == self.id:
-                self_index = i
-                break
-            i += 1
-
-        # Get the previous invoices if any.
-        previous_invoices = ordered_invoice_ids[:self_index]
-
-        # Get the incoming and outgoing sml between self.invoice_date and the previous invoice (if any) of the related product.
-        write_dates = [wd for wd in self.invoice_line_ids.mapped('write_date') if wd]
+        amls = self.invoice_line_ids.filtered(lambda aml: not aml.display_type and aml.product_id and aml.quantity)
+        write_dates = [wd for wd in amls.mapped('write_date') if wd]
```

Comment:
Please don't use the write_date to sort aml, it doesn't make sens, if you edit the invoice (for exemple the analityc account) the order is changed.

Please find a correct fix of this mecanism
https://github.com/odoo/odoo/pull/81444

Reply:
@adwid and make the difference if the unit of sml are not the same (and use the same unit of sml in the lot table): 
https://github.com/odoo/odoo/pull/81444/files#diff-82cd9c6ee43f18e312ed3364ec72b82930d06e72fb4f61cbb0bd0fffb936371aR80

case if you by  1 Tonne of product A and 250 grams product A.
you should see:
1 tonne : S/N : 765768987
250 grams : S/N : 554344445

and not:
1000 kg : S/N : 765768987
0,250 kg : S/N : 554344445

Reply:
Unfortunately, the mechanism from #814444 does not pass the test `test_backorder_and_several_invoices`

About the UoM, I don't completely agree: suppose several lines for the same lot L but with different UoM, I think that grouping them is more relevant. I let the stock-team decide

Reply:
About the UoM, it is not relevant for painting for exemple, it is possible to buy in different UoM the same lot, and if you have a mixing lot and UoM, it is hard to know what you are (it is a rare case).

case if you by 1 Tonne of product A and 250 grams product A from the same lot. And 250 grams from an other lot.
you should see:
1 tonne : S/N : 765768987
250 grams : S/N : 765768987
250 grams : S/N : 1342424

and not:
1000,25 kg : S/N : 765768987
250 grams : S/N : 1342424

Reply:
Where you see that https://github.com/odoo/odoo/pull/81444 doesn't pass the test ?
![image](https://user-images.githubusercontent.com/16716992/156159856-af52d026-79df-4ff4-b188-a2058e25b976.png)

Reply:
See the tests I added in this commit

Reply:
It is normal it fail because you use the write date (and my fix not). I can take you test file in my PR.

https://github.com/odoo/odoo/pull/85115/files#diff-0d062a0c602334c89ca4a4c92882d35dee9e4f01570545b087635b722d46e867R162

Reply:
No, the line you quote is used to make the test realistic
Do the procedure manually, directly on Odoo, you'll see that unfortunately, the use case doesn't work
`invoice_date` is not accurate enough: when printing the first invoice again, because both invoices have the same `invoice_date`, `previous_aml` will contain the AML of the second invoice -> this `previous_aml` will consume the first USN => the second USN will be displayed on the first invoice, which is incorrect
I didn't find a way to be more accurate than a date without using `write_date`